src/fetch/orderbook.py

```diff
@@ -45,17 +46,28 @@ def _pg_engine(db_env: OrderbookEnv) -> Engine:
         return create_engine(db_string)
 
     @classmethod
-    def _query_both_dbs(cls, query: str) -> tuple[DataFrame, DataFrame]:
-        barn = pd.read_sql(query, con=cls._pg_engine(OrderbookEnv.BARN))
-        prod = pd.read_sql(query, con=cls._pg_engine(OrderbookEnv.PROD))
+    def _read_query_for_env(
+        cls, query: str, env: OrderbookEnv, data_types: Optional[dict[str, str]] = None
+    ) -> DataFrame:
+        return pd.read_sql_query(query, con=cls._pg_engine(env), dtype=data_types)  # type: ignore
```

Comment:
I think there is a bug in mypy on the dtype field. They expect types that I believe match what we pass, but also don't provide a public interface to import their types (also np.int64 doesn't work either).

```
src/fetch/orderbook.py:51: error: Argument "dtype" to "read_sql_query" has incompatible type "Dict[str, str]"; expected "Optional[Union[Union[ExtensionDtype, Union[str, dtype[generic], Type[str], Type[complex], Type[bool], Type[object]]], Dict[Any, Union[ExtensionDtype, Union[str, dtype[generic], Type[str], Type[complex], Type[bool], Type[object]]]]]]"  [arg-type]
```
We ignore types on this line only for now.

Reply:
Okay, But I think its better practice to put this comment into the code.

Reply:
Done.

Reply:
I think the comment got too long. I will create an issue to put more context and link it in the comment.